client/app/bundles/course/reference-timelines/components/SeriouslyAnchoredPopup.tsx

```diff
@@ -0,0 +1,50 @@
+import { ComponentProps, useState } from 'react';
+import { Popover } from '@mui/material';
+
+interface AnchorPosition {
+  left: number;
+  top: number;
+}
+
+interface SeriouslyAnchoredPopupProps extends ComponentProps<typeof Popover> {
+  anchorEl: ComponentProps<typeof Popover>['anchorEl'];
+  anchorPosition?: never;
+  anchorReference?: never;
+  TransitionComponent?: never;
+  TransitionProps?: never;
+}
+
+/**
```

Comment:
This popup is not created to be used, rather to "address" a bug. There are two ways of anchoring a `Popover` (and other MUI `Portal` components):
1. The anchor (usually a button) sets `anchorEl` to itself from `event.currentTarget` `onClick`.
2. The developer manually sets `anchorEl` with a ref to the anchor (with `useRef`).

The problem with the first method is that, if the anchor element gets updated before `Popover` closes, `anchorEl` is not guaranteed to still be in the DOM, but its reference is still kept in state. When the `Popover` actually closes, it will re-render once (its own internal state implementation), and find that the given `anchorEl` reference actually resolves to `undefined` in the DOM, thus yielding `undefined` `left` and `top`, causing the `Popover` to jump to the top-left corner of the screen.

The second method would work if the DOM node is simply updated, but no guarantees across render cycles and/or unmounts.

The problem lies on the fact that `Popover` relies, _throughout its lifecycle_, on `anchorEl` for positioning. IMHO, this shouldn't be the case. The only reason why `Popover` gets `anchorEl` is to calculate the correct `left` and `top` to spawn itself. That's it. No need for `anchorEl` anymore when it is opening, opened, closing, or closed. `SeriouslyAnchoredPopup` effectively achieves this.